src/Compilers/CSharp/Portable/Symbols/UpdatedContainingSymbolLocal.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class UpdatedContainingSymbolAndNullableAnnotationLocal : LocalSymbol
+    {
+        internal UpdatedContainingSymbolAndNullableAnnotationLocal(SourceLocalSymbol underlyingLocal, Symbol updatedContainingSymbol, TypeWithAnnotations updatedType)
+        {
+            Debug.Assert(underlyingLocal is object);
+            Debug.Assert(updatedContainingSymbol is object);
+            UnderlyingLocal = underlyingLocal;
+            ContainingSymbol = updatedContainingSymbol;
+            TypeWithAnnotations = updatedType;
+        }
+
+        private SourceLocalSymbol UnderlyingLocal { get; }
+        public override Symbol ContainingSymbol { get; }
+        public override TypeWithAnnotations TypeWithAnnotations { get; }
+
+        public override bool Equals(Symbol other, TypeCompareKind compareKind)
+        {
+            if (other == (object)this)
+            {
+                return true;
+            }
+
+            return other is LocalSymbol local
+                && local.TypeWithAnnotations.Equals(TypeWithAnnotations, compareKind)
+                && local.ContainingSymbol.Equals(ContainingSymbol, compareKind)
+                && local.Equals(UnderlyingLocal, TypeCompareKind.AllNullableIgnoreOptions | TypeCompareKind.IgnoreTupleNames);
+        }
+
+        public override int GetHashCode()
+        {
+            return Hash.Combine(UnderlyingLocal,
```

Comment:
>return Hash.Combine(UnderlyingLocal, [](start = 12, length = 36)

I think we should simply delegate to the ```UnderlyingLocal```. Under some condition this symbol should be considered equal to the ```UnderlyingLocal```, therefore both should have the same hash code.